### DISCUSSION TITLE
How can I use Hasura with a GraphQL query to project the fields of the results into the required JSON format?

### DISCUSSION THREAD

**Discussion by AkshayBhatHasura** [hasura/graphql-engine#9789] [Question]
I am able to connect to two different databases from Hasura and query and obtain results, but the obtained JSON has to be tweaked to fit my needs.

**Comment by sumedh-1** [comment]
[Here is a reference to the same original query in Twitter](https://twitter.com/social__cop/status/1681684492552130560?s=20)

**Reply by code-sumedh** [reply]
I'm subscribing and participating in this discussion to chain this to my personal github @code-sumedh account as well.

**Reply by code-sumedh** [reply]
![image](https://github.com/hasura/graphql-engine/assets/32300690/c17666fb-3249-440e-9cfd-444c9faaa724)

**Reply by code-sumedh** [reply]
How much longer do I have to wait for someone to initiate the discussion with me?

**Reply by manasag** [reply]
Hi @code-sumedh, can you provide example of what kind of transformation you are aiming for. [Naming Conventions feature](https://hasura.io/docs/latest/schema/postgres/naming-convention/#supported-naming-conventions) provide some customizations around renaming fields.

**Reply by code-sumedh** [reply]
Hi @manasag,
 
My query is as below (queries two different databases using Hasura and obtain results):
![image](https://github.com/hasura/graphql-engine/assets/32300690/89dd677f-a7e7-4071-929d-812517195767)

and this is the response:
![image](https://github.com/hasura/graphql-engine/assets/32300690/f2302f93-8f9b-4a67-bed3-3e609dfa530d)

but the obtained JSON has to be tweaked to fit my needs as below:
![image](https://github.com/hasura/graphql-engine/assets/32300690/8c0b9854-e2f7-4a00-aac1-fca3ccc6efda)

**Reply by manasag** [reply]
Thanks for the details. I would say that this is a data union usecase; where we have same schema across different databases; and we want to collate results across different databases into a single graphql root field.

This is not supported in Hasura as of now. We think that in future that some of these usecases which can be very custom, can be solved by writing your own connector, in a easy to code against SDK in Hasura V3. V3 is due [beta launch](https://hasura.io/ddn) in August.  I would like to convert this discussion to an issue, so that we continue this discussion as a feature request and provide further updates.
